src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/iOS/WebUI.cs

```diff
@@ -44,6 +49,30 @@ public WebUI(IPlatformParameters parameters)
             {
                 throw new ArgumentException("parameters should be of type PlatformParameters", "parameters");
             }
+
+            DidEnterBackgroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.DidEnterBackgroundNotification, OnMoveToBackground);
+            WillEnterForegroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.WillEnterForegroundNotification, OnMoveToForeground);
+        }
+
+        void OnMoveToBackground(NSNotification notification)
+        {
+            taskId = UIApplication.SharedApplication.BeginBackgroundTask(() => {
+                inBackground = false;
+                UIApplication.SharedApplication.EndBackgroundTask(taskId);
```

Comment:
I'd suggest setting taskId to UIBackgroundTaskInvalid if there's such thing in Xamarin